lib/graph_gateway.go

```diff
@@ -352,23 +352,24 @@ func (api *GraphGateway) loadRequestIntoSharedBlockstoreAndBlocksGateway(ctx con
 				case blk, ok = <-blkCh:
 					if !t.Stop() {
 						<-t.C
-
 					}
 					t.Reset(GetBlockTimeout)
 				case <-t.C:
 					return gateway.ErrGatewayTimeout
 				}
-				if !ok {
+				if !ok || err != nil {
 					if errors.Is(err, io.EOF) {
 						return nil
 					}
 					return err
 				}
-				if err := bstore.Put(ctx, blk); err != nil {
-					return err
+				if blk != nil {
```

Comment:
@willscott 

1.Can `blk` ever be nil here ? How ?
2. Does it make sense to continue with the iteration if `blk` is nil here ?

Reply:
1. one hopes not
2. If we don't have an error, and have not finished the channel, it makes sense to drain it, right?